docs/en/sql-reference/statements/delete.md

```diff
@@ -7,57 +7,79 @@ keywords: [delete]
 title: DELETE Statement
 ---
 
+The `DELETE` statement removes rows from the table `[db.]table` that match the expression `expr`.
+
 ``` sql
-DELETE FROM [db.]table [ON CLUSTER cluster] WHERE expr
+DELETE FROM [db.]table [ON CLUSTER cluster] WHERE expr;
 ```
 
-`DELETE FROM` removes rows from the table `[db.]table` that match the expression `expr`. The deleted rows are marked as deleted immediately and will be automatically filtered out of all subsequent queries. Cleanup of data happens asynchronously in the background. This feature is only available for the MergeTree table engine family.
+Deleted rows are internally marked as deleted immediately and will be automatically filtered out of all subsequent queries. Cleanup of data happens asynchronously in the background until the next merge. As a result, it is possible that for an unspecified period, data is not actually deleted from storage and is only marked as deleted.
+
```

Comment:
Should we also somewhere at the top of the page mention and refer people to a "heavy" `ALTER table DELETE` query that will actually give you guarantees (asynchronously or synchronously) of deleting your data in a predictable period of time?